src/Payment.res

```diff
@@ -328,13 +328,17 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
   }, (cardNumber, cvcNumber, cardExpiry, isCVCValid, isExpiryValid, isCardValid))
 
   React.useEffect(() => {
-    let cardError = if isCardValid == None || cardNumber->String.length == 0 {
-      ""
-    } else if isCardSupported->Option.getOr(true) && isCardValid->Option.getOr(true) {
-      ""
-    } else if isCardSupported->Option.getOr(true) {
-      localeString.inValidCardErrorText
-    } else {
+    let cardError = switch (
+      isCardValid,
+      isCardSupported->Option.getOr(true),
+      isCardValid->Option.getOr(true),
+      cardNumber->String.length == 0,
+    ) {
+    | (None, _, _, _) => ""
+    | (_, _, _, true) => ""
+    | (_, true, true, _) => ""
+    | (_, true, _, _) => localeString.inValidCardErrorText
+    | (_, _, _, _) =>
```

Comment:
I feel the previous one is more readable and better. This is more confusing

Reply:
isCardValid seems redundant here, can we remove that? Also try to simplify this further

Reply:
Hey @aritro2002 

Try this -

```
  React.useEffect(() => {
    let cardError = if isCardValid == None || cardNumber->String.length == 0 {
      ""
    } else {
      let isCardSupportedValue = isCardSupported->Option.getOr(true)
      let isCardValidValue = isCardValid->Option.getOr(false)

      if isCardSupportedValue && isCardValidValue {
        ""
      } else if isCardSupportedValue {
        localeString.inValidCardErrorText
      } else {
        let cardBrand = cardNumber->CardUtils.getCardBrand
        if cardBrand == "" {
          localeString.inValidCardErrorText
        } else {
          localeString.cardBrandConfiguredErrorText(cardBrand)
        }
      }
    }
    setCardError(_ => cardError)
    None
  }, [isCardValid, isCardSupported])
```